src/clj/puppetlabs/services/master/master_core.clj

```diff
@@ -624,13 +556,68 @@
         (handler request)))))
 
 (schema/defn ^:always-validate
-  environment-class-handler :- IFn
-  "Handler for processing an incoming environment_classes Ring request"
-  [jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)
-   environment-class-cache-enabled :- schema/Bool]
-  (->
-   (environment-class-info-fn jruby-service
-                              environment-class-cache-enabled)
+  raw-transports->response-map
+  [data :- List
+   env :- schema/Str]
+  {:name env
+   :transports data})
+
+(schema/defn ^:always-validate
+  check-cache! :- ringutils/RingResponse
+  "Updates cache if new data does not match previous data
+
+  We must use the cache version generated at the start of the request.
+  The cache will check to see that the version given to it is the same
+  as its internal version (eg that it hasn't been reset since handing
+  out that version). If the given and internal versions do not match the
+  request to set the tag is ignored, if they match the tag is stored and
+  the internal version is incremented."
+  [info :- {schema/Any schema/Any}
+   env :- schema/Str
+   jruby-service :- (schema/protocol jruby-protocol/JRubyPuppetService)
+   request-tag :- (schema/maybe String)
+   cache-version :- (schema/maybe schema/Int)]
+  (let [body (cheshire/encode info)
+        tag (ks/utf8-string->sha1 body)]
```

Comment:
Do we have the same concerns about the SHA1 here @pcarlisle ?

Reply:
Since it's just for a cache I think it probably doesn't matter, though in cases like this if there's no cost to updating to sha256 I think we generally should.